Clans in nametags and scoreboard

Ageudum opened the pull request:
Revised some code from firestar's fork to implement colored clan-names
in player nametags and the scoreboard. Also fixed bug from firestar's
fork where player name and message in chat is dark red.

ChristopherJHart committed:
Clans in nametags and scoreboard

Revised some code from firestar's fork to implement colored clan-names
in player nametags and the scoreboard. Also fixed bug from firestar's
fork where player name and message in chat is dark red.

xtphty commented:
@firestar , @ChristopherJHart , @atlasesque

Need a way to make sure the tags are easily distinguishable in game.  Best way probably is to force clan tag colors to be unique, by assigning them randomly on clan creation (prefer), or just rejecting non unique colors (tedious for player).  

We can probably just enforce this at runtime, since names usually get picked in advance on teamspeak, but its best to automate things where possible.

Ageudum commented:
@xtphty ,

I agree. I'm working on a way to automatically assign colors upon clan creation as we speak - I'll be making a pull request (hopefully later tonight) that will resolve that issue.

ChristopherJHart committed:
Randomized new clan colors, clan-team interaction bugfix

Clantag colors are now randomly assigned from a pool of colors. If
player attempts to add on a color code at the beginning of the tag, it
is removed and replaced with an assigned one. Fixed bug where if a clan
is disbanded, the scoreboard was not updated appropriately.
